presto-main/src/test/java/io/prestosql/operator/scalar/time/TestExtract.java

```diff
@@ -156,4 +157,284 @@ public void testMillisecond()
         assertThat(assertions.expression("millisecond(TIME '12:34:56.12345678901')")).matches("BIGINT '123'");
         assertThat(assertions.expression("millisecond(TIME '12:34:56.123456789012')")).matches("BIGINT '123'");
     }
+
+    @Test
+    public void testYear()
+    {
+        assertThatThrownBy(() -> assertions.expression("EXTRACT(YEAR FROM TIME '12:34:56')"))
+                .isInstanceOf(PrestoException.class)
+                .hasMessage("line 1:26: Cannot extract YEAR from time(0)");
+
+        assertThatThrownBy(() -> assertions.expression("EXTRACT(YEAR FROM TIME '12:34:56.1')"))
+                .isInstanceOf(PrestoException.class)
+                .hasMessage("line 1:26: Cannot extract YEAR from time(1)");
+
+        assertThatThrownBy(() -> assertions.expression("EXTRACT(YEAR FROM TIME '12:34:56.12')"))
```

Comment:
in https://github.com/prestosql/presto/pull/5614 i decided to test negative cases for one precision only. i guess it could be a bit simplistic... no strong opinion

you could test the negative case with a loop for brevity:

```
for precision 0 to 12
 assert "EXTRACT .. FROM CAST(NULL AS time(precision))" fails 
```

(message guarantees the failure is NULL-unrelated)

Reply:
> in #5614 i decided to test negative cases for one precision only. i guess it could be a bit simplistic... no strong opinion

It's important to check all. Otherwise, it's easy for bugs due to `type.equals(xxx)` to slip by.

Reply:
> you could test the negative case with a loop for brevity:

Let me give that a try. I tend to prefer the explicit cases because it's 1) easier to step over the code, comment out parts, place breakpoints, etc. to debug specific issues. 2) the tests are more self-evident (although I understand this is subjective)